contrib/hashicorp/vault/vault_test.go

```diff
@@ -0,0 +1,312 @@
+		mt := mocktracer.Start()
+		defer mt.Stop()
+		defer mountKV(c, t)()
+
+		// Write key
+		_, err := c.Logical().Write(key, data)
+		if err != nil {
+			t.Fatal(err)
+		}
+		spans := mt.FinishedSpans()
+		assert.Len(spans, 2)
+		span := spans[1]
+
+		assert.Equal("vault", span.Tag(ext.ServiceName))
+		assert.Equal(fullPath, span.Tag(ext.HTTPURL))
+		assert.Equal(http.MethodPut, span.Tag(ext.HTTPMethod))
+		assert.Equal(http.MethodPut+" "+fullPath, span.Tag(ext.ResourceName))
+		assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+		assert.Equal(200, span.Tag(ext.HTTPCode))
+		assert.Zero(span.Tag(ext.Error))
+		assert.Zero(span.Tag("vault.namespace"))
+	})
+
+	t.Run("read", func(t *testing.T) {
+		mt := mocktracer.Start()
+		defer mt.Stop()
+		defer mountKV(c, t)()
+
+		// Write the key first
+		_, err := c.Logical().Write(key, data)
+		if err != nil {
+			t.Fatal(err)
+		}
+		// Read key
+		secret, err := c.Logical().Read(key)
+		if err != nil {
+			t.Fatal(err)
+		}
+		spans := mt.FinishedSpans()
+		assert.Len(spans, 3)
+		span := spans[2]
+
+		assert.Equal(secret.Data["Key1"], data["Key1"])
+		assert.Equal(secret.Data["Key2"], data["Key2"])
+		assert.Equal("vault", span.Tag(ext.ServiceName))
+		assert.Equal(fullPath, span.Tag(ext.HTTPURL))
+		assert.Equal(http.MethodGet, span.Tag(ext.HTTPMethod))
+		assert.Equal(http.MethodGet+" "+fullPath, span.Tag(ext.ResourceName))
+		assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+		assert.Equal(200, span.Tag(ext.HTTPCode))
+		assert.Zero(span.Tag(ext.Error))
+		assert.Zero(span.Tag("vault.namespace"))
+	})
+}
+
+func TestReadError(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	ts, cleanupTs := setupServer(t)
+	defer cleanupTs()
+	client, err := setupClient(ts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer mountKV(client, t)()
+
+	key := "/some/bad/key"
+	fullPath := "/v1" + key
+	secret, err := client.Logical().Read(key)
+	if err == nil {
+		t.Fatalf("Expected error when reading key from %s, but it returned: %#v", key, secret)
+	}
+	spans := mt.FinishedSpans()
+	assert.Len(spans, 2)
+	span := spans[1]
+
+	// Read key error
+	assert.Equal("vault", span.Tag(ext.ServiceName))
+	assert.Equal(fullPath, span.Tag(ext.HTTPURL))
+	assert.Equal(http.MethodGet, span.Tag(ext.HTTPMethod))
+	assert.Equal(http.MethodGet+" "+fullPath, span.Tag(ext.ResourceName))
+	assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+	assert.Equal(404, span.Tag(ext.HTTPCode))
+	assert.NotZero(span.Tag(ext.Error))
+	assert.Zero(span.Tag("vault.namespace"))
+}
+
+func TestNamespace(t *testing.T) {
+	assert := assert.New(t)
+
+	ts, cleanupTs := setupServer(t)
+	defer cleanupTs()
+	client, err := setupClient(ts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer mountKV(client, t)()
+
+	namespace := "/some/namespace"
+	client.SetNamespace(namespace)
+	key := secretMountPath + "/testNamespace"
+	fullPath := "/v1" + key
+	t.Run("write", func(t *testing.T) {
```

Comment:
I think `t.Run` declaration do warrant an empty line above them. It's nice to see each unit test as a separate unit. Like you did in `testMountReadWrite` 👍

Reply:
👍